app/queries/edition_links.rb

```diff
@@ -34,57 +46,83 @@ def call
 
   private
 
-    attr_reader :content_id, :mode, :locale, :with_drafts, :allowed_link_types
+    attr_reader :content_id,
+                :mode,
+                :locale,
+                :with_drafts,
+                :allowed_link_types,
+                :parent_content_ids,
+                :next_allowed_link_types_from,
+                :next_allowed_link_types_to
 
     def initialize(
       content_id:,
       mode:,
       locale:,
       with_drafts:,
-      allowed_link_types:
+      allowed_link_types:,
+      parent_content_ids: [],
+      next_allowed_link_types_from: nil,
+      next_allowed_link_types_to: nil
     )
       @content_id = content_id
       @mode = mode
       @locale = locale
       @with_drafts = with_drafts
       @allowed_link_types = allowed_link_types
+      @parent_content_ids = parent_content_ids
+      @next_allowed_link_types_from = next_allowed_link_types_from
+      @next_allowed_link_types_to = next_allowed_link_types_to
     end
 
     def links_results
-      query = Link.left_joins(edition: :document)
-      where(query)
-        .order(link_type: :asc, position: :asc)
-        .pluck(:link_type, link_field, "documents.locale", "editions.id")
-    end
+      condition = {}
+      # condition[:"documents.locale"] = locale if locale
+      condition[:link_type] = allowed_link_types if allowed_link_types
 
-    def link_field
-      mode == :from ? :target_content_id : "documents.content_id"
+      if mode == :from
+        puts "from"
+        Link
+          .left_joins(edition: :document)
+          .left_joins(:link_set)
```

Comment:
You shouldn't need to join link_set here. If you look at the way this works in root_links_by_link_type, the idea is to make separate queries - one for link set links, and one for edition links.

```
  def root_links_by_link_type(content_id:, locale:, with_drafts: false)
    direct = own_links(content_id)
    reverse = linked_to(content_id, rules.reverse_links)
    edition = edition_links(content_id, locale, with_drafts)

    reverse.merge(direct).merge(edition)
  end
```

I assume we'd want to take a similar approach in `child_links_by_link_type`.